lib/engine/game/base.rb

```diff
@@ -607,12 +620,17 @@ def initialize(names, id: 0, actions: [], at_action: nil, pin: nil, strict: fals
         @log << '----'
       end
 
+      def initialize_seed(seed)
+        @seed = (seed || @id.to_s.scan(/\d+/).first.to_i) % RAND_M
```

Comment:
Should the seed itself be modulo'd by `RAND_M` or only @rand?

Reply:
When I did that I thought there would be no difference since the operations in `rand` apply the mod operation, but the turn order shuffling done on the ruby side does use the seed without modding it. So it's a minor difference, but doesn't hurt to move it to `@rand`, done.